07-DevOps/01-devops-astropy.md

```diff
@@ -0,0 +1,179 @@
+  more powerful but harder to use.
+
+Other services out there (list not exhaustive):
+
+* [GitLab CI](https://docs.gitlab.com/ee/ci/) for CI support like GitHub Actions
+  but for code hosted on GitLab.
+* [Jenkins](https://www.jenkins.io/) that is useful if you have a lot of
+  internal stuff and have the expertise to understand/maintain this framework.
+  This is not really recommended for small project with limited resources.
+
+Complementary services/tools that are useful but has limited scope
+(list not exhaustive):
+
+* [ReadTheDocs](https://about.readthedocs.com/) for online documentation building, hosting, and
+  [pull request previews](https://docs.readthedocs.io/en/stable/pull-requests.html).
+* [pre-commit.ci](https://pre-commit.ci/) for style checks if you use
+  [pre-commit](https://pre-commit.com/).
+* [Codecov](https://about.codecov.io/) for code coverage reporting.
+* [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) for testing against
+  edge cases, especially in low-level library code.
+
+## How do I decide what to use?
+
+Software is a fast-moving domain. The services mentioned above (or any you
+come across in the wild) can change overnight. This is a risk you must
+accept when picking a service that you do not have complete control over.
+Case studies:
+
+* [Travis CI was acquired by Idera](https://techcrunch.com/2019/01/23/idera-acquires-travis-ci/)
+  and started to monetize even the open source projects. Astropy had to
+  scramble to move its reliance to GitHub Actions. Some others chose to pay.
+* [Anaconda looking to monetize even non-profit institutional usage](https://www.theregister.com/2024/08/08/anaconda_puts_the_squeeze_on/)
+  causing some projects including Astropy to move away to alternatives like
+  [miniforge](https://github.com/conda-forge/miniforge).
+  (This situation is still ongoing at the time of writing.)
+
+With this in mind, make a list of "must, must not, and nice to have" features
+and discuss with your team which tools best fit your needs. There is no
+one right answer.
+
+# What does DevOps look like in Astropy?
+
+[![Astropy Infrastructure](https://private-user-images.githubusercontent.com/2090236/337734652-9cfe3833-61f1-46bb-8507-172ce117ec73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI0ODczNTgsIm5iZiI6MTczMjQ4NzA1OCwicGF0aCI6Ii8yMDkwMjM2LzMzNzczNDY1Mi05Y2ZlMzgzMy02MWYxLTQ2YmItODUwNy0xNzJjZTExN2VjNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjRUMjIyNDE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWFiZmZlMTQ3ZTZiZTA3ZGU0YzA3NWE5NTdjNzBlZTNlZTRkYTE3ZGIxZTNiMTY4ZTBkZjZkMjVjNmE0MzFmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SkTUxGruah5Jdksh_ZvDlHmErrI5CQGz8bgmj_KSr_0)](https://zenodo.org/records/11521582)
```

Comment:
FYI, this is not displaying inline for me...

Reply:
It worked last night. I wonder if they rotate the token in URL. 😿 

I'll see if I can get something more permanent. Thanks for pointing that out!

Reply:
Can't find a direct link to Zenodo version either. I get an annoying "viewer" instead. So I just added the file here. Does it work now? Is this solution acceptable?